src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcSocket.cs

```diff
@@ -23,6 +22,8 @@ public IpcSocket(AddressFamily addressFamily, SocketType socketType, ProtocolTyp
         {
         }
 
+// .NET 6 implements this method directly on Socket, but for earlier runtimes we need a polyfill
+#if !NET6_0
         public async Task<Socket> AcceptAsync(CancellationToken token)
```

Comment:
FYI - this override will always be used at runtime given our compilation modes, regardless of the runtime version that gets used.